frodo.js

```diff
@@ -1,58 +1,57 @@
 /**
```

Comment:
SCRIPT-I'm impressed with your use of design patterns here, it really enhances the code organization.